### PR TITLE
fix "return statement uses an exception to pass control ..." warning

### PR DESCRIPTION
**Pull request by jtjeferreira**
fix "return statement uses an exception to pass control to the caller of the enclosing named method"

### CODE REVIEW

**Comment by typesafe-tools** [comment]
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild/?sbt=develop&zinc=pull/891/head&io=develop&librarymanagement=develop&util=develop&website=develop&deploy_to_bintray=false&run_sbt_community=false&verbose_dbuild=false&community_verbose=false">this link.</A>

**Review by eed3si9n — APPROVED** [review]

File: internal/compiler-bridge/src/main/scala/xsbt/Dependency.scala [hunk 1]
```diff
@@ -72,13 +72,14 @@ final class Dependency(val global: CallbackGlobal) extends LocateClassFile with
...
-          val sym = classOrModule.symbol
-          return Some(if (sym.isModule) sym.moduleClass else sym)
-        case _ => ()
+      val maybeClassOrModule = tree find {
+        case ((_: ClassDef) | (_: ModuleDef)) => true
+        case _                                => false
+      }
+      maybeClassOrModule.map { classOrModule =>
+        val sym = classOrModule.symbol
+        if (sym.isModule) sym.moduleClass else sym
```
Comment (dwijnand) [inline internal/compiler-bridge/src/main/scala/xsbt/Dependency.scala:81]: collect over find/map...

Comment (jtjeferreira) [reply internal/compiler-bridge/src/main/scala/xsbt/Dependency.scala:81]: I think you meant `collectFirst`, however `collectFirst` does not exist for `Tree`

Comment (dwijnand) [reply internal/compiler-bridge/src/main/scala/xsbt/Dependency.scala:81]: Ah, fair enough (and yeah, collectFirst 😳)

**Pull request merged**